clippy_lints/src/methods.rs

```diff
@@ -1263,8 +1263,12 @@ fn lint_map_unwrap_or(cx: &LateContext, expr: &hir::Expr, map_args: &[hir::Expr]
     // lint if the caller of `map()` is an `Option`
     if match_type(cx, cx.tables.expr_ty(&map_args[0]), &paths::OPTION) {
         // get snippets for args to map() and unwrap_or()
-        let map_snippet = snippet(cx, map_args[1].span, "..");
-        let unwrap_snippet = snippet(cx, unwrap_args[1].span, "..");
+        let map_span = map_args[1].span;
+        let unwrap_span = unwrap_args[1].span;
+        let map_and_unwrap_span = map_args[0].span.next_point().with_hi(expr.span.hi());
```

Comment:
This is surely not the best way to do it…

Reply:
I think the best way would be to use a multi-part suggestion. One part would be for the method name, the other for inserting the `default_value, ` before the closure.